compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -950,15 +969,181 @@ impl<'a> PartialEvaluator<'a> {
+                                .expect("array length should fit into u32"),
+                        )),
+                        Operand::Literal(Literal::Pointer),
+                    ],
+                    None,
+                ));
+                for val in vals.iter() {
+                    instrs
+                        .extend(self.generate_output_recording_instructions(val.clone(), elem_ty));
+                }
+            }
+            Value::Tuple(vals) => {
+                let Ty::Tuple(elem_tys) = ty else {
+                    panic!("expected tuple type for tuple value");
+                };
+                let tuple_record_callable_id = self.get_tuple_record_callable();
+                instrs.push(Instruction::Call(
+                    tuple_record_callable_id,
+                    vec![
+                        Operand::Literal(Literal::Integer(
+                            vals.len()
+                                .try_into()
+                                .expect("tuple length should fit into u32"),
+                        )),
+                        Operand::Literal(Literal::Pointer),
+                    ],
+                    None,
+                ));
+                for (val, elem_ty) in vals.iter().zip(elem_tys.iter()) {
+                    instrs
+                        .extend(self.generate_output_recording_instructions(val.clone(), elem_ty));
+                }
+            }
+
+            Value::Result(res) => {
+                let result_record_callable_id = self.get_result_record_callable();
+                instrs.push(Instruction::Call(
+                    result_record_callable_id,
+                    vec![
+                        Operand::Literal(Literal::Result(
+                            res.unwrap_id()
+                                .try_into()
+                                .expect("result id should fit into u32"),
+                        )),
+                        Operand::Literal(Literal::Pointer),
+                    ],
+                    None,
+                ));
+            }
+            Value::Var(var) => {
+                let (record_callable_id, record_ty) = match ty {
+                    Ty::Prim(Prim::Bool) => (self.get_bool_record_callable(), rir::Ty::Boolean),
+                    Ty::Prim(Prim::Int) => (self.get_int_record_callable(), rir::Ty::Integer),
+                    _ => panic!("unsupported variable type in output recording"),
+                };
+                instrs.push(Instruction::Call(
+                    record_callable_id,
+                    vec![
+                        Operand::Variable(Variable {
+                            variable_id: var.0.into(),
+                            ty: record_ty,
+                        }),
+                        Operand::Literal(Literal::Pointer),
+                    ],
+                    None,
+                ));
+            }
+
+            Value::BigInt(_)
```

Comment:
@swernli I thought that we implicitly wrapped single values of supported types in a tuple.

Reply:
No, if only a single value is returned then it doesn't have any aggregate and will show up as it's own idividual item.